src/app/pages/datasets/components/data-protection-card/data-protection-card.component.ts

```diff
@@ -13,15 +18,39 @@ import { IxSlideInService } from 'app/services/ix-slide-in.service';
   templateUrl: './data-protection-card.component.html',
   styleUrls: ['./data-protection-card.component.scss'],
 })
-export class DataProtectionCardComponent implements OnInit {
+export class DataProtectionCardComponent implements OnInit, OnChanges {
   readonly console = console;
-  @Input() dataset: DatasetDetails;
-
+  changeEvent$: Subject<string> = new Subject();
+  isLoading = false;
   constructor(
     private slideIn: IxSlideInService,
     private snackbarService: SnackbarService,
     private translate: TranslateService,
-  ) {}
+    private ws: WebSocketService,
+    private cdr: ChangeDetectorRef,
+  ) {
+    // Counting the number of snapshots for zvol
+    // https://ixsystems.atlassian.net/browse/NAS-118753
+    this.changeEvent$.pipe(
+      switchMap((id) => {
+        this.isLoading = true;
+        return this.ws.call('zfs.snapshot.query', [[['dataset', '=', id]]]);
+      }),
+      untilDestroyed(this),
+    ).subscribe((res) => {
+      this.dataset = { ...this.dataset, snapshot_count: res.length };
+      this.isLoading = false;
+      this.cdr.markForCheck();
+    });
+  }
+
+  @Input() dataset: DatasetDetails;
```

Comment:
let's move it at the top before `readonly console = console;`
and add empty line